### PR TITLE
8265349: vmTestbase/../stress/compiler/deoptimize/Test.java fails with OOME due to CodeCache exhaustion.

### PR DESCRIPTION
**Pull request by lepestock**
The bug: https://bugs.openjdk.java.net/browse/JDK-8265349

A repetition of the JDK-8058176 (mlvm tests cause code cache exhaustion), this time with -Xcomp. My measurements show up max code cache consumption of 400-500 kb per test thread and tree (depends on random tree and other factors, of course, but still). For the whole test which usually doesn't exceed 10 threads, I've got max. value of 6.1M for one sequence/tree build (between checks for remaining free space).

So I suggest to raise the allowances to 10M, just to be safe. Compared to the 103 Mb of required code cache space that test requests via the 'run' parameter, it doesn't look that much.

Thanks in advance,
// Evgeny Nikitin.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265349](https://bugs.openjdk.java.net/browse/JDK-8265349): vmTestbase/../stress/compiler/deoptimize/Test.java fails with OOME due to CodeCache exhaustion.

### Reviewers
 * [Igor Ignatyev](https://openjdk.java.net/census#iignatyev) (@iignatev - **Reviewer**) ⚠️ Review applies to c3f52fbfc48cb511f507ab4cf4ec0ff5382d061a

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3762/head:pull/3762` \
`$ git checkout pull/3762`

Update a local copy of the PR: \
`$ git checkout pull/3762` \
`$ git pull https://git.openjdk.java.net/jdk pull/3762/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3762`

View PR using the GUI difftool: \
`$ git pr show -t 3762`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3762.diff">https://git.openjdk.java.net/jdk/pull/3762.diff</a>

</details>

### CODE REVIEW

**Comment by iignatev** [comment]
Will it make sense to make `NON_SEGMENTED_CACHE_ALLOWANCE` / `SEGMENTED_CACHE_ALLOWANCE` value dependent on `Platform::isComp`?

**Comment by lepestock** [comment]
> Will it make sense to make `NON_SEGMENTED_CACHE_ALLOWANCE` / `SEGMENTED_CACHE_ALLOWANCE` value dependent on `Platform::isComp`?

Please take a look at the new version. I measured code cache usage without the Xcomp, and it falls within 1-2MB per cycle/tree with 10 threads setup (I've did several runs). I suggest 5M in this case, just to be sure we'll not hit some high-amplitude resonance.

**Review by iignatev — CHANGES_REQUESTED** [review]
I don't like that we kinda guessing here, but on the other hand, I don't think we have much of the choice.

could you please add the comment which roughly explains why it's 10kk and 5kk? so the next person who reads/modifies this piece of code will have an idea of how/why we get the current numbers.

**Comment by lepestock** [comment]
> I don't like that we kinda guessing here, but on the other hand, I don't think we have much of the choice.
> 
> could you please add the comment which roughly explains why it's 10kk and 5kk? so the next person who reads/modifies this piece of code will have an idea of how/why we get the current numbers.

I've added an explanation and extracted the code cache monitoring in a sub-class (because it had lots of members and explanations dedicated only to this functionality).

File: test/hotspot/jtreg/vmTestbase/vm/mlvm/meth/share/MHTransformationGen.java [hunk 1]
```diff
@@ -68,45 +70,66 @@
...
+
+        // Trial runs show up that maximal increase in code cache consumption between checks (for one
+        // cycle/tree build in MHTransformationGen::createSequence), falls within the following intervals:
+        //
+        // | Threads number | Without Xcomp | With Xcomp |
+        // |----------------|---------------|------------|
+        // |       1        |   100-200 K   |  400-500 K |
+        // |      10        |    1 - 2 M    |    5-6 M   |
+        //
+        // Those numbers are approximate (since threes are generated randomly and the total consumption
```
Comment (iignatev) [inline test/hotspot/jtreg/vmTestbase/vm/mlvm/meth/share/MHTransformationGen.java:92]: s/threes/trees/ ?

**Review by iignatev — APPROVED** [review]
other than a typo, LGTM.

File: test/hotspot/jtreg/vmTestbase/vm/mlvm/meth/share/MHTransformationGen.java (hunk 1 above)
Comment (lepestock) [reply test/hotspot/jtreg/vmTestbase/vm/mlvm/meth/share/MHTransformationGen.java]: > s/threes/trees/ ?

How did you know that these were not actual threes / troikas? :))

Fixed, thanks.

**Pull request closed without merging**